### ISSUE TITLE
insert before regression

### ISSUE THREAD

**Issue by idoros** [postcss/postcss#1785]
In version `8.4.17` there was some regression when inserting new nodes with `before()`. Prior to this patch version using the `before()` method would add the nodes with a newline between the reference node and the new node. 

see new lines when adding AST nodes in `8.4.16`: [example](https://codesandbox.io/s/musing-morse-d7h4bs?file=/src/index.js)
and no newlines when doing the same in `8.4.17`: [example](https://codesandbox.io/s/priceless-estrela-7mzdbm?file=/src/index.js)

**Comment by romainmenke** [comment]
I will try to take a look this weekend.

**Comment by idoros** [comment]
Fair enough, as long as you are aware of these changes, feel free to close.

**Comment by massivelines** [comment]
For me the whitespace is important, we use `postcss`, `postcss-scss`, and `jscodeshit` for migrations in our design system.

When 
```scss
@import '@thezebra/styles/src/scss/style';
// test comment
```
turns into this it can cause us some issues. We run a linter after the transformation but that still won't move the comment back down to the original line.
```scss
@import '@thezebra/styles/src/scss/style';// test comment
```

**Comment by ai** [comment]
@massivelines do you want to find a solution which will fix the origin issue without changing whitespace filling algorithm?

**Comment by romainmenke** [comment]
@massivelines I had to take some time to remain focussed on another effort but your report came at just the right time!

@ai sorry it took so long for me to get around to this.
I've opened a pull request with a fix for this issue.

**Comment by ai** [comment]
Fixed by @romainmenke in 8.4.19
